### PR TITLE
Rewrite and redirect canonical topic aliases.

### PR DESCRIPTION
**Pull request by isoos**
#7263

### CODE REVIEW

**Review by jonasfj — APPROVED** [review]

File: app/lib/package/model_properties.dart [hunk 1]
```diff
@@ -198,7 +204,7 @@ class Pubspec {
   }
 
   late final List<Uri> funding = _inner.funding ?? const <Uri>[];
-  late final hasTopic = topics?.isNotEmpty ?? false;
+  late final hasTopic = canonicalizedTopics.isNotEmpty;
```
[the remark this answers is not in the document]
Comment (isoos) [reply app/lib/package/model_properties.dart]: I think both names are valid (as long as it is consistent with other parts of the code). IIRC we don't really have much similar use cases, or we don't have `hasX[s]` methods for them, so we may just choose any variant. I don't mind either way, any preference?

File: app/lib/search/backend.dart [hunk 2]
```diff
@@ -522,6 +523,30 @@ class SearchBackend {
   }
 }
 
+/// Returns a new search form that may overrides predicates to their canonical forms.
```
Comment (szakarias) [inline app/lib/search/backend.dart:526]: overrides -> override

File: pkg/_pub_shared/lib/search/search_form.dart [hunk 3]
```diff
@@ -123,6 +123,24 @@ class TagsPredicate {
     return p;
   }
 
+  /// Update the predicates by using their canonical version (if there is any available).
+  /// Does not change the predicates status (unless the canonical one is also present,
+  /// in this case the behavior is undefinited).
```
Comment (szakarias) [inline pkg/_pub_shared/lib/search/search_form.dart:128]: undefinited -> undefined

**Review by szakarias — APPROVED** [review]
Nice!

**Pull request merged**